### PR TITLE
Stabilize deconvolution2d test

### PR DESCRIPTION
**Pull request by okuta**

### CODE REVIEW

File: tests/chainer_tests/links_tests/connection_tests/test_deconvolution_2d.py [hunk 1]
```diff
@@ -150,7 +150,7 @@ def check_backward(self, x_data, y_grad):
             params.append(self.link.b)
 
         gradient_check.check_backward(
-            self.link, x_data, y_grad, params, eps=1e-2)
+            self.link, x_data, y_grad, params, eps=1e-2, rtol=1e-3, atol=1e-4)
```
Comment (unnonouno) [inline tests/chainer_tests/links_tests/connection_tests/test_deconvolution_2d.py]: Why not use `dtype='d'`

Comment (okuta) [reply tests/chainer_tests/links_tests/connection_tests/test_deconvolution_2d.py]: Because this is connection test.

Comment (niboshi) [reply tests/chainer_tests/links_tests/connection_tests/test_deconvolution_2d.py]: >Because this is connection test.

What does this mean?

**Comment by niboshi** [comment]
Could you fix `TestDeconvolution2D` too? Its backward is still unstable.
Also please remove `retry`.

File: tests/chainer_tests/links_tests/connection_tests/test_deconvolution_2d.py (hunk 1 above)
Comment (okuta) [reply tests/chainer_tests/links_tests/connection_tests/test_deconvolution_2d.py]: Sorry, Because this is link test.

**Comment by okuta** [comment]
OK, I will

File: tests/chainer_tests/links_tests/connection_tests/test_deconvolution_2d.py (hunk 1 above)
Comment (niboshi) [reply tests/chainer_tests/links_tests/connection_tests/test_deconvolution_2d.py]: Well, why don't link tests need `dtype`?
In my understanding, underlying numerical computations are not different between link tests and function tests.

Comment (okuta) [reply tests/chainer_tests/links_tests/connection_tests/test_deconvolution_2d.py]: Sorry, I was misunderstanding. I will fix

Comment (niboshi) [reply tests/chainer_tests/links_tests/connection_tests/test_deconvolution_2d.py]: OK!

**Comment by niboshi** [comment]
jenkins, test this please

**Comment by niboshi** [comment]
It seems backward test is still unstable only when cuDNN is enabled.

**Comment by okuta** [comment]
I fixed.

**Comment by niboshi** [comment]
LGTM!

**Pull request merged**